File: cmd/models/task.go
package models

import "time"

type Task struct {
	ID          string     `json:"id"`
	ColumnID    string     `json:"columnId"`
	Title       string     `json:"title"`
	Description *string    `json:"description"`
	CreatorID   string     `json:"creatorId"`
	AssigneeID  *string    `json:"assigneeId"`
	Assignee    *User      `json:"assignee,omitempty"`
	Priority    string     `json:"priority"`
	Deadline    *time.Time `json:"deadline"`
	Position    int        `json:"position"`
	Subtasks    []Subtask  `json:"subtasks"`
	Labels      []Label    `json:"labels"`
	CreatedAt   time.Time  `json:"createdAt"`
	UpdatedAt   time.Time  `json:"updatedAt"`
	CompletedAt *time.Time `json:"completedAt"`
}

type Subtask struct {
	ID          string    `json:"id"`
	TaskID      string    `json:"taskId"`
	Title       string    `json:"title"`
	IsCompleted bool      `json:"isCompleted"`
	CreatedAt   time.Time `json:"createdAt"`
}

type Label struct {
	ID      string `json:"id"`
	BoardID string `json:"boardId"`
	Name    string `json:"name"`
	Color   string `json:"color"`
}

File: app/src/types/task.ts
import type { User } from './user'

export type Priority = 'urgent' | 'high' | 'medium' | 'low'

export interface Task {
  id: string
  columnId: string
  title: string
  description: string | null
  creatorId: string
  assigneeId: string | null
  assignee: User | null
  assignees: User[]
  priority: Priority
  deadline: string | null
  position: number
  subtasks: Subtask[]
  labels: Label[]
  createdAt: string
  updatedAt: string
  completedAt: string | null
}

export interface Subtask {
  id: string
  taskId: string
  title: string
  isCompleted: boolean
  createdAt: string
}

export interface Label {
  id: string
  boardId: string
  name: string
  color: string
}

export interface TaskDelegation {
  id: string
  taskId: string
  fromUserId: string
  toUserId: string
  reason: string
  delegatedAt: string
}

export interface Comment {
  id: string
  taskId: string
  userId: string
  user: User
  content: string
  createdAt: string
}

export interface ActivityLog {
  id: string
  taskId: string
  userId: string
  user: User
  action: string
  details: Record<string, unknown>
  createdAt: string
}

export const PRIORITY_CONFIG: Record<Priority, { label: string; color: string; icon: string }> = {
  urgent: { label: 'Urgent', color: '#dc2626', icon: 'mdi-alert-circle' },
  high: { label: 'High', color: '#06b6d4', icon: 'mdi-arrow-up-bold' },
  medium: { label: 'Medium', color: '#6366f1', icon: 'mdi-minus' },
  low: { label: 'Low', color: '#64748b', icon: 'mdi-arrow-down-bold' },
}
